packages/hooks/src/createRunHook.js

```diff
@@ -0,0 +1,47 @@
+/**
+ * Returns a function which, when invoked, will execute all callbacks
+ * registered to a hook of the specified type, optionally returning the final
+ * value of the call chain.
+ *
+ * @param  {Object}   hooks          Stored hooks, keyed by hook name.
+ * @param  {?bool}    returnFirstArg Whether each hook callback is expected to
+ *                                   return its first argument.
+ *
+ * @return {Function}                Function that runs hook callbacks.
+ */
+function createRunHook( hooks, returnFirstArg ) {
+	/**
+	 * Runs all callbacks for the specified hook.
+	 *
+	 * @param  {string} hookName The name of the hook to run.
+	 * @param  {...*}   args     Arguments to pass to the hook callbacks.
+	 *
+	 * @return {*}               Return value of runner, if applicable.
+	 */
+	return function runHooks( hookName, ...args ) {
+		const handlers = hooks[ hookName ];
+		let maybeReturnValue = args[ 0 ];
+
+		if ( ! handlers ) {
```

Comment:
Along lines of previous comment about separating `handlers`, since `removeAll` flow in `removeHook` will reset handlers to an empty array, this condition won't trigger. Would we expect `runs` to increment after all handlers are removed for a hook? Otherwise this would at least need to be changed to `if ( ! handlers || ! handlers.length )` (since empty array is truthy).